lib/metadata/schema-parser.js

```diff
@@ -1071,6 +1078,33 @@ function getTokenToReplicaNetworkMapper(replicationFactors) {
 }
 
 /**
+ * @returns {Number} The number of skipped hosts added.
+ */
+function addSkippedHosts(dcRf, dcReplicas, tokenReplicas, skippedHosts) {
+  var i;
+  for (i = 0; i < dcRf - dcReplicas && i < skippedHosts.length; i++) {
+    tokenReplicas.push(skippedHosts[i]);
+  }
+  return i;
+}
+
+function isDoneForToken(replicationFactors, datacenters, replicasByDc) {
+  var keys = Object.keys(replicationFactors);
+  for (var i = 0; i < keys.length; i++) {
+    var dcName = keys[i];
+    var dc = datacenters[dcName];
+    if (!dc) {
+      // A DC is included in the RF but the DC does not exist in the topology
+      continue;
+    }
+    var rf = Math.min(parseInt(replicationFactors[dcName], 10), dc.hostLength);
+    if (replicasByDc[dcName] < rf) {
```

Comment:
This is not perfect since it needs to consider the case where there are 0 replicas belonging to a DC.  Checking hostLength for 0 explicitly should short circuit that.